### DISCUSSION TITLE
What is eclairs road map regarding (negative) inbound fees?

### DISCUSSION THREAD

**Discussion by DerEwige** [ACINQ/eclair#2853] [Node operators]
LND just released their first release candidate for version 0.18.0.
This version includes the ability to configure inbound fees on channels.
Per default it is only possible to set negative inbound fees, to keep compatibility with nodes that do not understand the new fees. 

Nodes who do not understand those new negative inbound fees will simply overpay when using that route.

But in the long run this would mean that those nodes would start to overpay a lot in the best case or get a lot more failed payments in the worst case (when nodes start to use positive inbound fees)

I have not seen any PR or fork related to this new fee and was wondering how the eclair road map looks regarding this?

**Comment by t-bast** [comment]
We don't have plans to support inbound fees in the short term. There were discussions between implementations, and two competing bLIP were proposed:

- https://github.com/lightning/blips/pull/18
- https://github.com/lightning/blips/pull/22

I believe lnd implemented the first one, while our suggestion was to use the second one instead. At that point, it's unclear whether inbound fees are a good idea at all (there was a lot of push back for the use-cases advertised) and how it should be done. So we'll simply wait for more discussions to happen once people start getting real-world experience with that feature.

**Comment by DerEwige** [comment]
hey @t-bast I just wanted to give a short update on the current state.

Lnd v0.18.0 release about 2 weeks ago.
The version seems to be rather stable and no major bug was found so far.
People have already started upgrading and confidence is rising.
I expect the number of lnd 0.18.0 nodes to rise fast in the next few weeks.

The master branches of some tools (lndg, charge-lnd) already have support for the inbound fees.

Currently there are already about 500 channels in the network that offer negative inbound fees.
(pleas see attached document [inbound_fees.csv](https://github.com/user-attachments/files/15793849/inbound_fees.csv))

So fare no one has set positive inbound fees (this would break backwards compatibility)

I will ask my friend for an updated list every few weeks to let you know about the current state.

**Comment by DerEwige** [comment]
short update:

First 2 nodes have started to use positive inbound fees, which makes those channels incompatible with eclair.

Nodes with TLV stream 55555 set: 93
Nodes with at least one negative channel: 45
Nodes with at least one positive channel: 2
Channels  with TLV stream 55555 set: 2927
Negative Inbound Fee channels: 703
Postitive Inbound Fee channels: 32
